integration-tests/bats/garbage_collection.bats

```diff
@@ -476,3 +469,114 @@ SQL
+    mkdir -p one/two
+    cd one/two
+    dolt init
+    rm -rf .dolt/stats
+    ls -laR > ../../before_gc
+    dolt gc --full
+    rm -rf .dolt/stats
+    ls -laR > ../../after_first_gc
+    cp ./.dolt/noms/manifest ../../manifest_after_first_gc
+    dolt gc --full
+    rm -rf .dolt/stats
+    ls -laR > ../../after_second_gc
+    # This should exit 0 because the gc should have changed things.
+    if cmp ../../before_gc ../../after_first_gc; then
+        echo "expected dolt gc to change things, but it didn't."
+        diff ../../before_gc ../../after_first_gc || true
+        false
+    fi
+    # This should exit non-0 because the gc should NOT have changed things.
+    if ! cmp ../../after_first_gc ../../after_second_gc || ! cmp ./.dolt/noms/manifest ../../manifest_after_first_gc; then
+        echo "expected dolt gc --full after a dolt gc --full to not change things, but it did."
+        diff ../../after_first_gc ../../after_second_gc || true
+        false
+    fi
+}
+
+@test "garbage_collection: dolt gc after dolt gc --full is a no-op" {
+    mkdir -p one/two
+    cd one/two
+    dolt init
+    rm -rf .dolt/stats
+    ls -laR > ../../before_gc
+    dolt gc --full
+    rm -rf .dolt/stats
+    ls -laR > ../../after_first_gc
+    cp ./.dolt/noms/manifest ../../manifest_after_first_gc
+    dolt gc
+    rm -rf .dolt/stats
+    ls -laR > ../../after_second_gc
+    # This should exit 0 because the gc should have changed things.
+    if cmp ../../before_gc ../../after_first_gc; then
+        echo "expected dolt gc to change things, but it didn't."
+        diff ../../before_gc ../../after_first_gc || true
+        false
+    fi
+    # This should exit non-0 because the gc should NOT have changed things.
+    if ! cmp ../../after_first_gc ../../after_second_gc || ! cmp ./.dolt/noms/manifest ../../manifest_after_first_gc; then
+        echo "expected dolt gc after a dolt gc --full to not change things, but it did."
+        diff ../../after_first_gc ../../after_second_gc || true
+        false
+    fi
+}
+
+@test "garbage_collection: dolt gc --full after dolt gc is NOT a no-op" {
+    mkdir -p one/two
+    cd one/two
+    dolt init
+    rm -rf .dolt/stats
+    ls -laR > ../../files_before_gc
+    dolt gc
+    rm -rf .dolt/stats
+    ls -laR > ../../files_after_first_gc
+    cp ./.dolt/noms/manifest ../../manifest_after_first_gc
+    dolt gc --full
+    rm -rf .dolt/stats
+    ls -laR > ../../files_after_second_gc
+    # This should exit 0 because the gc should have changed things.
+    if cmp ../../files_before_gc ../../files_after_first_gc; then
+        echo "expected dolt gc to change things, but it didn't."
+        diff ../../files_before_gc ../../files_after_first_gc || true
+        false
+    fi
+    # This should exit non-0 because the gc should NOT have changed things.
+    if cmp ../../files_after_first_gc ../../files_after_second_gc && cmp ./.dolt/noms/manifest ../../manifest_after_first_gc; then
```

Comment:
Do we expect both the files and the manifest to change or just one of them? We should break this into two separate checks.

Reply:
In reality, they can both change, or only the manifest can change. But the manifest must change.

The file listing for this particular case only changes if the order that the chunks are written to the stores is different or if the created table files as a result of the GC operation have two different create timestamps because the operations are run in different minutes.

So best to check for the lack of differences in both in the no-change case, and sufficient to check for a change in just the manifest in this case. So I've changed it to just look at the manifest.